mtags/src/main/scala-2/scala/meta/internal/pc/PcRenameProvider.scala

```diff
@@ -0,0 +1,31 @@
+package scala.meta.internal.pc
+
+import scala.meta.internal.mtags.MtagsEnrichments._
+import scala.meta.pc.OffsetParams
+
+import org.eclipse.lsp4j.TextEdit
+
+class PcRenameProvider(
+    override val compiler: MetalsGlobal,
+    params: OffsetParams,
+    name: String
+) extends PcCollector[TextEdit](compiler, params) {
+  import compiler._
+  val newName: String = Identifier.backtickWrap(name.stripBackticks)
+  def collect(tree: Tree, pos: Position): TextEdit = {
+    val isBackticked = text(pos.start) == '`' && text(pos.end - 1) == '`'
+    // when the old name contains backticks, the position is incorrect
+    val isOldNameBackticked = text(pos.start) == '`' &&
+      text(pos.end - 1) != '`' &&
+      text(pos.end + 1) == '`'
+    if (isBackticked)
+      new TextEdit(pos.toLsp, "`" + newName.stripBackticks + "`")
+    else if (isOldNameBackticked)
+      new TextEdit(pos.withEnd(pos.end + 2).toLsp, newName)
+    else new TextEdit(pos.toLsp, newName)
```

Comment:
```suggestion
    if (isOldNameBackticked)
      new TextEdit(pos.withEnd(pos.end + 1).withStart(pos.start - 1).toLsp, newName)
    else new TextEdit(pos.toLsp, newName)
```
doesn't seem that we need to strip backticks and add them again, no? Also `+2` doesn't seem correct. Could you make sure that the behaviour is consistent in VS Code and in tests?

Btw. it seems we should handle backtick in `PcCollector`, since semantic highlighting also is having issues. So ideally we would adjust the position `pos` before giving it to the collect function.

Reply:
I've added a `adjust` which handles backticks. I tested it thoroughly and the behaviour is the same as in nonlocal rename. `+2` does look suspicious , but is needed (before highlight on `val 'great!'` was `val  <<'great>>!'`).